docs/architecture/maui/communicating-between-components.md

```diff
@@ -1,115 +1,115 @@
-    });
+WeakReferenceMessenger.Default
+    .Register<CatalogView, Messages.AddProductMessage>(
+        this,
+        async (recipient, message) =>
+        {
+            await recipient.Dispatcher.DispatchAsync(
+                async () =>
+                {
+                    await recipient.badge.ScaleTo(1.2);
+                    await recipient.badge.ScaleTo(1.0);
+                });
+        });
 ```
 
-In the preceding example, the Subscribe method subscribes to the `AddProduct` message, and executes a callback delegate in response to receiving the message. This callback delegate, specified as a lambda expression, executes code that updates the UI.
+In the preceding example, the `Register` method subscribes to the `AddProductMessage` message and executes a callback delegate in response to receiving the message. This callback delegate, specified as a lambda expression, executes code that updates the UI.
 
-If payload data is supplied, don't attempt to modify the payload data from within a callback delegate because several threads could be accessing the received data simultaneously. In this scenario, the payload data should be immutable to avoid concurrency errors.
+> [!NOTE]
+> Avoid the use of `this` within your callback delegate to avoid capturing that object within the delegate. This can help improve performance. Instead, use the `recipient` parameter.
 
-A subscriber might not need to handle every instance of a published message, and this can be controlled by the generic type arguments that are specified on the Subscribe method. In this example, the subscriber will only receive `AddProduct` messages that are sent from the `CatalogViewModel` class.
+If payload data is supplied, don't attempt to modify the payload data from within a callback delegate because several threads could be accessing the received data simultaneously. In this scenario, the payload data should be immutable to avoid concurrency errors.
 
 ## Unsubscribing from a message
 
-Subscribers can unsubscribe from messages they no longer want to receive. This is achieved with one of the `MessagingCenter`.Unsubscribe overloads, as demonstrated in the following code example:
+Subscribers can unsubscribe from messages they no longer want to receive. This is achieved with one of the `IMessenger.Unregister` overloads, as demonstrated in the following code example:
 
 ```csharp
-MessagingCenter.Unsubscribe<CatalogViewModel>(
-    this, MessengerKeys.AddProduct);
+WeakReferenceMessenger.Default.Unregister<Messages.AddProductMessage>(this);
 ```
 
-In this example, the Unsubscribe method syntax reflects the type arguments specified when subscribing to receive the AddProduct message.
+> [!NOTE]
+> In this example, it is not fully necessary to call `Unregister` as the `WeakReferenceMessenger` will allow unused objects to be garbage collected. If the `StrongReferenceMessenger` were used, it would be advised to call `Unregister` for any subscriptions that are no longer in use.
```

Comment:
```suggestion
> In this example, it isn't fully necessary to call `Unregister` as the `WeakReferenceMessenger` will allow unused objects to be garbage collected. If the `StrongReferenceMessenger` were used, it would be advised to call `Unregister` for any subscriptions that are no longer in use.
```